### DISCUSSION TITLE
Get custom answer in arg() prompt

### DISCUSSION THREAD

**Discussion by johnstonmatt** [johnlindquist/kit#845] [Q&A] [answered]
I like the behaviour of `await arg("Pick a favourite fruit:", ["Apple", "Banana"])` as it exists, but is there a way (other function or param I'm not seeing) to have the same API, with the difference being that if the user continues to type after there are no results matching the fuzzy filter (ie: "Watermelon"), we get this custom text as a return value when they tap `enter` to submit?

**Comment by johnlindquist** [comment, marked as answer]
## Disable `strict`

Yup, the first arg of `arg` can also be a config object. On the config, set `strict` to `false` then it will allow you submit any arbitrary value.


[Open strict-mode-demo in Script Kit](https://scriptkit.com/api/new?name=strict-mode-demo&url=https://gist.githubusercontent.com/johnlindquist/7eba1c8fdc9557018b230ddd62b7a8ff/raw/8b5e3d9e10f113d1b8b730496d35592acc6ed3e1/strict-mode-demo.js")

```js
// Name: Strict Mode Demo

import "@johnlindquist/kit"

let fruit = await arg({
    placeholder: "Select a fruit",
    hint: "Type 'Grape' and hit enter",
    strict: false
}, ["Apple", "Banana", "Cherry"])

await div(md(fruit))
```

## Manual Submit

If you want to get more advanced, you can call `submit()` from anywhere to advance the current active prompt:

[Open manual-submit-demo in Script Kit](https://scriptkit.com/api/new?name=manual-submit-demo&url=https://gist.githubusercontent.com/johnlindquist/834856692c63340ed5ba6508f1d42673/raw/eadbbb4ef67e8facb1a049d47f1ab838f888121d/manual-submit-demo.js")

```js
// Name: Manual Submit Demo

import "@johnlindquist/kit"

let argPromise = arg("Waiting for value")

setTimeout(() => { 
    submit({message: "whoa..."})
}, 2000)

let value = await argPromise

await div(md(value.message))
```
